includes/Revision.php

```diff
@@ -295,20 +295,14 @@ private static function loadFromConds( $db, $conditions, $flags = 0 ) {
 	}
 
 	/**
-	 * Return a wrapper for a series of database rows to
-	 * fetch all of a given page's revisions in turn.
-	 * Each row can be fed to the constructor to get objects.
-	 *
-	 * @param $title Title
-	 * @return ResultWrapper
+	 * @param $row
 	 */
-	public static function fetchRevision( $title ) {
-		return self::fetchFromConds(
-			wfGetDB( DB_SLAVE ),
-			array( 'rev_id=page_latest',
-				   'page_namespace' => $title->getNamespace(),
-				   'page_title'     => $title->getDBkey() )
-		);
+	private static function replaceUsernameFieldsWithVariables( $row )
+	{
+		$user = User::newFromId( $row->rev_user );
```

Comment:
`rev_user` can be zero. In such cases it's an anon edit and rev_user_text has IP address. We should use `User::getUsername` here instead that handles such cases.